src/NHibernate/Linq/Visitors/QueryModelVisitor.cs

```diff
@@ -162,56 +156,39 @@ public override void VisitMainFromClause(MainFromClause fromClause, QueryModel q
 
 		public override void VisitAdditionalFromClause(AdditionalFromClause fromClause, QueryModel queryModel, int index)
 		{
-			var querySourceName = VisitorParameters.QuerySourceNamer.GetName(fromClause);
-
-			var joinClause = fromClause as NhJoinClause;
-			if (joinClause != null)
-			{
-				VisitNhJoinClause(querySourceName, joinClause);
-			}
-			else if (fromClause.FromExpression is MemberExpression)
+			if (fromClause.FromExpression is MemberExpression)
```

Comment:
Here, I have to admit I have left the test cases check for me that apparently, `NhJoinClause` were never used with something not castable to `MemberExpression`. Otherwise, some cases previously handled with `NhJoinClause` but not using a `MemberExpression` will no more work.

It looks to me reasonable to do such an assumption since HQL permit joins only through member references as far as I know.